### DISCUSSION TITLE
Yosys optimization

### DISCUSSION THREAD

**Discussion by salmen20** [YosysHQ/yosys#4346] [General]
Hello,
hello i am facing a problem as i want to use yosys for synth but i dont want for any optimization to happen , tried removing the makefile tried everything but nothing worked.
any idea how can i achieve that ?
Thanks.

**Comment by KrystalDelusion** [comment]
https://yosyshq.readthedocs.io/projects/yosys/en/latest/getting_started/example_synth.html guides one through the process of synthesis.  You can try follow along and skip the steps you don't need.  Depending on what platform you are trying to synthesize for you can also look at the relevant `synth_*` pass.  The guide runs through `synth_ice40` but the general process is the same for all of them.

You shouldn't mess with the makefile unless you know what you are doing.  It should be possible to do what you want with an unmodified Yosys executable.  Once you figure out the steps you need, you can then make a script (there's a [docs page for that](https://yosyshq.readthedocs.io/projects/yosys/en/latest/getting_started/scripting_intro.html) too) to repeat the process.
